src/celeritas/ext/g4vg/Converter.cc

```diff
@@ -159,45 +230,56 @@ auto Converter::build_with_daughters(G4LogicalVolume const* mother_g4lv)
 
     ++depth_;
 
+    auto convert_daughter = [this](G4LogicalVolume const* g4lv) {
+        return this->build_with_daughters(g4lv);
+    };
+
     // Place daughter logical volumes in this mother
-    for (auto const i : range(mother_g4lv->GetNoDaughters()))
+    for (auto i : range(mother_g4lv->GetNoDaughters()))
     {
         // Get daughter volume
-        G4VPhysicalVolume const* g4pv = mother_g4lv->GetDaughter(i);
-        G4LogicalVolume const* g4lv = g4pv->GetLogicalVolume();
-        if (!dynamic_cast<G4PVPlacement const*>(g4pv))
-        {
-            TypeDemangler<G4VPhysicalVolume> demangle_pv_type;
-            CELER_LOG(error)
-                << "Unsupported type '" << demangle_pv_type(*g4pv)
-                << "' for physical volume '" << g4pv->GetName()
-                << "' (corresponding LV: " << PrintableLV{g4lv} << ")";
-        }
+        G4VPhysicalVolume* g4pv = mother_g4lv->GetDaughter(i);
 
-        // Test for reflection
-        bool flip_z = false;
-        if (G4LogicalVolume const* unrefl_g4lv = get_constituent_lv(*g4lv))
+        if (auto* placed = dynamic_cast<G4PVPlacement const*>(g4pv))
         {
-            // Replace with constituent volume, and flip the Z scale
-            // See G4ReflectionFactory::CheckScale: the reflection value is
-            // hard coded to {1, 1, -1}
-            g4lv = unrefl_g4lv;
-            flip_z = true;
+            DaughterPlacer place_daughter(convert_daughter,
+                                          *convert_transform_,
+                                          g4pv->GetLogicalVolume(),
+                                          mother_lv);
+
+            // Place daughter, accounting for reflection
+            place_daughter(placed);
         }
+        else if (auto* div = dynamic_cast<G4PVDivision*>(g4pv))
```

Comment:
looking good to me overall. Maybe one could generalize from G4PVDivision to G4PVReplica ?(although I am no real expert on this).

Reply:
Looks like "replica" doesn't necessarily mean "parameterized", but that's a good idea so I've generalized it a bit.